specification/servicefabric/resource-manager/Microsoft.ServiceFabric/preview/2023-11-01-preview/cluster.json

```diff
@@ -0,0 +1,2167 @@
+          {
+            "value": "Hierarchical",
+            "description": "If this value is omitted or set to Hierarchical, VMs are grouped to reflect the zonal distribution in up to 15 UDs. Each of the three zones has five UDs. This ensures that the zones are updated one at a time, moving to next zone only after completing five UDs within the first zone. This update process is safer for the cluster and the user application."
+          }
+        ]
+      },
+      "description": "This property controls the logical grouping of VMs in upgrade domains (UDs). This property can't be modified if a node type with multiple Availability Zones is already present in the cluster."
+    },
+    "StoreName": {
+      "type": "string",
+      "description": "The local certificate store location.",
+      "enum": [
+        "AddressBook",
+        "AuthRoot",
+        "CertificateAuthority",
+        "Disallowed",
+        "My",
+        "Root",
+        "TrustedPeople",
+        "TrustedPublisher"
+      ]
+    },
+    "UpgradableVersionPathResult": {
+      "type": "object",
+      "properties": {
+        "supportedPath": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "The cluster code version."
+          }
+        }
+      },
+      "description": "The list of intermediate cluster code versions for an upgrade or downgrade. Or minimum and maximum upgradable version if no target was given"
+    },
+    "UpgradableVersionsDescription": {
+      "type": "object",
+      "required": [
+        "targetVersion"
+      ],
+      "properties": {
+        "targetVersion": {
+          "type": "string",
+          "description": "The target code version."
+        }
+      }
+    },
+    "UpgradeMode": {
+      "type": "string",
+      "description": "The upgrade mode of the cluster when new Service Fabric runtime version is available.",
+      "enum": [
+        "Automatic",
+        "Manual"
+      ],
+      "default": "Automatic",
+      "x-ms-enum": {
+        "name": "UpgradeMode",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Automatic",
+            "description": "The cluster will be automatically upgraded to the latest Service Fabric runtime version, **upgradeWave** will determine when the upgrade starts after the new version becomes available."
+          },
+          {
+            "value": "Manual",
+            "description": "The cluster will not be automatically upgraded to the latest Service Fabric runtime version. The cluster is upgraded by setting the **clusterCodeVersion** property in the cluster resource."
+          }
+        ]
+      }
+    },
+    "VMSize": {
+      "type": "object",
+      "properties": {
+        "size": {
+          "type": "string",
+          "description": "VM Size name.",
+          "readOnly": true
+        }
+      },
+      "description": "VM Sizes properties."
+    },
+    "VMSizeResource": {
```

Comment:
>VMSizeResource [](http://example.com/codeflow?start=5&length=14)

please use ProxyResource definition from common types to define the top level common properties

Reply:
For common-types we have work planned for this https://msazure.visualstudio.com/One/_workitems/edit/25890469, But it's going to take some time because we generate the swagger from an internal repo that currently can't reference the common types.